### ISSUE TITLE
ProgressBar overflows if BorderThickness is used

### ISSUE THREAD

**Issue by MagnificentEagle** [AvaloniaUI/Avalonia#4489]
Hello!
I noticed a rather strange bug with the ProgressBar.

To find this problem, I subscribed to changing the text in the TextBox to calculate the complexity of the password and output in the ProgressBar, ``` Maximum = 300 ```.
With ``` BorderThickness = 0 ``` and ``` CornerRadius = 5 ```, everything is fine, working as expected.
![image](https://user-images.githubusercontent.com/44552715/90317515-89726580-df43-11ea-8ec5-38c8d239e244.png)
But if you set BorderThickness to 1, you get this:
![n29sBTeyES](https://user-images.githubusercontent.com/44552715/90317452-1cf76680-df43-11ea-840e-6dcba94550a9.gif)

**Comment by SCLD-JMcCoard** [comment]
Duplicated, but with extra details.

**When you have a border thickness set (higher the worse things get):**
* The precision position is offset (by the left border) 
* Also, the right side is clipped off.
* And some corner (radius) clipping.

Saw a report on it in gitter from "Printer", but because of any sensitivity for users on having the progress being true/exact, I'd thought I'd drop a note here.

```
<StackPanel VerticalAlignment="Center" HorizontalAlignment="Center" Orientation="Vertical">
    <TextBlock HorizontalAlignment="Center">|</TextBlock>
    <ProgressBar Height="30" Width="300" Minimum="0" Maximum="10" Value="5" BorderThickness="5" BorderBrush="Black" Foreground="SteelBlue"/>
    <TextBlock HorizontalAlignment="Center">|</TextBlock>
</StackPanel>
```

A possible **_work around_** is to wrap the progressbar in a border and use that border. I think the PART_Indicator doesn't really account for it's parents' border.  

```
<StackPanel VerticalAlignment="Center" HorizontalAlignment="Center" Orientation="Vertical">
     <TextBlock HorizontalAlignment="Center">|</TextBlock>
     <Border BorderThickness="2" BorderBrush="Black" CornerRadius="5">
          <ProgressBar Height="30" Width="300" Minimum="0" Maximum="10" Value="5" Foreground="SteelBlue"/>
     </Border>
     <TextBlock HorizontalAlignment="Center">|</TextBlock>
</StackPanel>
```

**Comment by maxkatz6** [comment]
Yes, PART_Indicator should use its parent bounds instead of total progress bar bounds.
